.github/workflows/os-extension-test.yml

```diff
@@ -154,6 +154,8 @@ jobs:
     needs: build
     steps:
       - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha || github.ref }}
```

Comment:
Maybe it's better to use simply `${{ github.ref }}` - the difference is, that "github.event.pull_request.head.sha" is the head commit of the PR, but "github.ref" will be the "pull request merge branch" - that is a temporary merge of the PR into main. This is usually, what we want, as we want to test the integration of the PR with the main branch. Otherwise we might merge the PR but the project will be broken, because the merged PR was never tested.

If this is changed here, then it must also be changed on line 42.